### DISCUSSION TITLE
Single sheet

### DISCUSSION THREAD

**Discussion by alexruiz95** [KRproject-tech/FSI_by_FEM_and_UVLM#3] [General]
Hello,

This is an awesome code great work. I have a question. I am going through the code and trying to see how I can modify it to have a single sheet. Is it easily possible?

**Comment by KRproject-tech** [comment]
It is time-consuming to correct this code to a single-sheet case. 
Therefore, I will upload the program for a single sheet to GitHub later.

**Reply by alexruiz95** [reply]
Thank you for your response! That would be much appreciated. 

Also would it be possible to do something like this with a single sheet? This is kind of my goal for the code. Clamped on one side and a displacement applied along the span? or some other boundary condition. 

![bendingandtwsting](https://github.com/KRproject-tech/FSI_by_FEM_and_UVLM/assets/39660318/6dcc8f71-8371-4a74-b882-f15f709f022e)

**Reply by KRproject-tech** [reply]
Analytical conditions are in "./XXXX/save/param_setting.m"
A boundary condition with one edge clamped and the other edge free is set as
----------------------------------------------------------------------------------------------------------------------------
%%[0] 0値代入
node_r_0 = [ 1:Ny+1:(Nx+1)*(Ny+1)];                                       %% Node number giving the displacement constraint [-]
node_dxr_0 = [ 1:Ny+1:(Nx+1)*(Ny+1)];                                     %% Node number giving x-directional gradient constraint [-]
node_dyr_0 = [ 1:Ny+1:(Nx+1)*(Ny+1)];                                     %% Node number giving y-directional gradient constraint [-]
----------------------------------------------------------------------------------------------------------------------------

**Reply by alexruiz95** [reply]
Thanks for the upload. I read through the code more and I modified those nodes for the boundary conditions. 

Depending on what is easier, is there a way to add a time dependent displacement? Either as field function or as a list of displacements as a function of time.  I would like to start the simulation and after some time bend or twist like the animation above and then return to initial position. 

I been reading through the code and translating the comments but can't seem to determine where I would do this.

I appreciate any help you can provide on this!

Thanks again.

**Reply by KRproject-tech** [reply]
Perhaps it is possible to give displacement as a boundary condition by modifying "new_X_func_FAST.m" in the following procedure
[Time dependent clamped BC.pdf](https://github.com/KRproject-tech/FSI_by_FEM_and_UVLM/files/11553917/Time.dependent.clamped.BC.pdf)

**Reply by alexruiz95** [reply]
Hmm.. Looking at new_X_func_FAST.m. I think it would be this line `out_0 = out1.A1_A2_Xn + A_mat1\( d_t*[ zero_mat(:,1);
                                       Q_global]);`

of this section? 


`X2_matrix = [ zero_mat      eye_mat;
              zero_mat      zero_mat];
A_mat1 = D_matrix - alpha_v*d_t*X2_matrix;
A_mat2 = D_matrix + (1 - alpha_v)*d_t*X2_matrix;

%%[2-0] 固定部のノードでは dt_q = 0, dtt_q = 0 [2-0] dt_q = 0, dtt_q = 0 at the nodes of the fixed part
not_i_vec = (1:N_q_all);
not_i_vec(i_vec) = [];

%% [3] 状態ベクトル更新 [3] State vector update

if stage == 0                                                                                           %% 1回目の計算値を使いまわす．%% Reuse the first calculated value.
    out1.A1_A2_Xn = A_mat1\( A_mat2*X_vec([ not_i_vec N_q_all+not_i_vec]) );
end
out_0 = out1.A1_A2_Xn + A_mat1\( d_t*[ zero_mat(:,1);
                                       Q_global]);
                    
out = X_vec;
out([ not_i_vec N_q_all+not_i_vec]) = out_0;
out = blkdiag( Fmat, Fmat)*out;

end`

Im looking at the variables and values the sim produces and what new_X_func_FAST.m produces.

**Reply by alexruiz95** [reply]
Additionally i have two questions,
1. what is the typical compute time on your end? It takes a long time on 16 cores. 
2. Do you happen to have essentially the same code with no structure? like just the UVLM?  Since I only care for the flow field and aerodynamic coefficients induced by the bending and twisting a one way FSI. 

I think this code will go along way with my research if I can adapt it.

Thanks for all the help and feedback it is really appreciated.

**Reply by KRproject-tech** [reply]
>1. what is the typical compute time on your end? It takes a long time on 16 cores.
Currently checking with Ryzen 5900X.

>2. Do you happen to have essentially the same code with no structure? like just the UVLM? Since I only care for the flow field and aerodynamic coefficients induced by the bending and twisting a one way FSI.
Added mass term in the pressure acting on a sheet is modified from the force calculation given by Katz & Plotkin to achieve strong two-way coupling between the fluid and a sheet.

**Reply by alexruiz95** [reply]
Hi,

Yeah I am curious on the compute time on your end. Sometimes things work fast on one computer but not the other. Is it supposed to be super fast? For a 2sec run for a single sheet it takes 1009 seconds. 

Still struggling on the time dependent displacement. Perhaps a similar question that could help me work it out is how would I apply a displacement from the start? Im looking at how we apply the displacement of zero, how would I apply at other nodes a small displacement?

Thanks for all the feedback

**Reply by KRproject-tech** [reply]
The times employed in the numerical analysis is non-dimensionalised by length and flow velocity as 
![image](https://github.com/KRproject-tech/FSI_by_FEM_and_UVLM/assets/114337358/86589d79-0ac6-4c44-b80e-7b42a995b30c)

The computing time under Ryzen 5900X is 2.5 hour for t^* = 30 [-].
